Hyperion/Extensions/TypeEx.cs

```diff
@@ -200,24 +203,27 @@ public static int GetTypeSize(this Type type)
 
         public static string GetShortAssemblyQualifiedName(this Type type)
```

Comment:
This is not requirement, but an idea: `GetShortAssemblyQualifiedName` is actually used in 2 external places (`TypeSerializer` and `ObjectSerializer`). In both places this string is converted to a byte array immediately after. What we could do here, is to make this method return a byte array directly, avoiding string conversions in the future.